encfs/RawFileIO.cpp

```diff
@@ -225,23 +226,25 @@ ssize_t RawFileIO::write(const IORequest &req) {
   off_t offset = req.offset;
 
   /*
-   * Let's write while pwrite() writes, to avoid writing only a part of the request,
-   * whereas it could have been fully written. This to avoid inconsistencies / corruption.
+   * Let's write while pwrite() writes, to avoid writing only a part of the
+   * request,
+   * whereas it could have been fully written. This to avoid inconsistencies /
+   * corruption.
    */
   // while ((bytes != 0) && retrys > 0) {
   while (bytes != 0) {
     ssize_t writeSize = ::pwrite(fd, buf, bytes, offset);
 
-    if (writeSize <= 0) {
+    if (writeSize < 0) {
       int eno = errno;
-      errno = 0; //just to be sure error seen in integration tests really comes from the function rc.
       knownSize = false;
       RLOG(WARNING) << "write failed at offset " << offset << " for " << bytes
                     << " bytes: " << strerror(eno);
-      // pwrite is not expected to return 0, so eno should always be set, but we never know...
-      if (eno) {
-      	return -eno;
-      }
+      // pwrite is not expected to return 0, so eno should always be set, but we
+      // never know...
+      return -eno;
+    }
+    if (writeSize == 0) {
       return -EIO;
     }
```

Comment:
Here we return `EIO` we loose a chance to return correct `errno` : depending on implementations `pwrite` could set `errno` when it returns 0.
This is why I tested `if (writeSize <= 0)` [in my initial commit](https://github.com/vgough/encfs/blob/acbeb647658a334ff00b2f2b1dac2addb2082bad/encfs/RawFileIO.cpp#L235).
Could we then get back to previous version here ?
Many thanks 👍